lib/puppet/file_bucket/dipper.rb

```diff
@@ -56,6 +58,38 @@ def backup(file)
     end
   end
 
+  # Diffs two filebucket files identified by their sums
+  def diff(suma, sumb)
+    raise RuntimeError, "Diff is not supported on windows platforms" if Puppet.features.microsoft_windows?
```

Comment:
It'd be preferable to disable this only when `Puppet[:diff]` isn't set (which is the default on Windows). If a diff method is configured, it should be used. That requires modifying the indirector to use `Puppet[:diff]` instead of `diff` below.